src/frontend/stmclient.cc

```diff
@@ -126,7 +127,7 @@ void STMClient::init( void )
   if ( (escape_key_env = getenv( "MOSH_ESCAPE_KEY" )) != NULL ) {
     if ( strlen( escape_key_env ) == 1 ) {
       escape_key = (int)escape_key_env[0];
-      if ( escape_key > 0 && escape_key < 128 ) {
+      if ( escape_key > 0 || escape_key < 128 ) {
```

Comment:
And in addition to one character ascii value, empty environment (explicitly set to empty string) disables the escape character altogether. Environment of any other kind, is just ignored.